index.js

```diff
@@ -77,59 +77,50 @@ const styles = {
   bgWhiteBright: [107, 49]
 };
 
-const isString = val => val && typeof val === 'string';
-const unstyle = val => isString(val) ? val.replace(ansiRegex, '') : val;
-const hasOpen = (input, open) => input.slice(0, open.length) === open;
-const hasClose = (input, close) => input.slice(-close.length) === close;
-const color = (str, style, hasNewline) => {
-  const { open, close } = style;
-  if (!(hasOpen(str, open) && hasClose(str, close))) {
-    str = style.open + str.replace(style.closeRe, style.open) + style.close;
-  }
-  if (hasNewline) {
-    return str.replace(/(\r?\n)/g, `${style.close}$1${style.open}`);
-  }
-  return str;
-};
-
-function wrap(...args) {
-  if (colors.visible === false) return '';
+function wrap() {
+  if (!colors.visible) return '';
+  let out = format.apply(null, arguments);
+  if (!colors.enabled || out.trim().length === 0) return out;
 
-  let str = util.format(...args);
-  if (colors.enabled === false) return str;
-  if (str.trim() === '') return str;
+  let i=0, tmp, arr=this.stack;
+  let isMulti = !!~out.indexOf('\n');
 
-  const newline = str.includes('\n');
-  for (const key of this.stack) {
-    str = color(str, styles[key], newline);
+  for (; i < arr.length; i++) {
+    tmp = styles[arr[i]]; // { x1, x2, rgx }
+    out = tmp.open + out.replace(tmp.closeRe, tmp.open) + tmp.close;
+    isMulti && (out = out.replace(/(\r?\n)/g, `${tmp.close}$1${tmp.open}`));
   }
-  return str;
+
+  return out;
 }
 
 function style(stack) {
-  const create = (...args) => wrap.call(create, ...args);
```

Comment:
This line was also costly. It's much cheaper to bind a new function reference than it is to create a new one entirely. Avoiding the spread operator is also generally a good idea